assets/php/virion.php

```diff
@@ -193,3 +198,36 @@ function change_dna(string $chromosome, string $antigen, string $antibody, $mode
 
     return $ret;
 }
+
+final class TempDir {
+
+    /** @var string */
+    public $path;
+
+    public function __construct() {
+        $this->path = tempnam(sys_get_temp_dir(), "virion-inject");
+        unlink($this->path);
+        $this->path .= DIRECTORY_SEPARATOR;
+        mkdir($this->path, 0666);
+    }
+
+    public function addFile(string $path, string $data): void {
+        $absolutePath = $this->path . $path;
+        if(file_exists($absolutePath)) throw new ErrorException("File $absolutePath already exists");
+        $absoluteDirPath = (new \SplFileInfo($absolutePath))->getPath();
```

Comment:
What does this function do?

Reply:
It gets the `"/path/to"` from `"/path/to/file.php"`, I think i'll shift this to `pathinfo($absolutePath)["dirname"]` instead.

Reply:
What's wrong with using `dirname`?

Reply:
Ah, i wasn't aware of that function, will use that